### ISSUE TITLE
Bare return detected in an anynomous function inside a function only if the latter function’s return value is named

### ISSUE THREAD

**Issue by olivierperes** [mgechev/revive#280]
**Describe the bug**
With `[rule.bare-return]` on, Revive emits an error message in the case detailed below, and not in another case, also detailed below, that I think should be equivalent.

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following flags & configuration file:

```shell
# flags

revive ...
```

```toml
[rule.bare-return]
```

**Expected behavior**
No output.

**Logs**
`main.go:5:3: avoid using bare returns, please add return expressions`

**Desktop (please complete the following information):**
 - OS: Linux Mint 19.2 (based on Ubuntu 18.04)
 - Version of Go: 1.13.4

**Additional context**

Here is the minimum example that triggers the Revive error:
```go
package main

func f() (err error) {
	func() {
		return
	}()

	return nil
}
```

No errors are produced if the return value is not named:
```go
package main

func f() error {
	func() {
		return
	}()

	return nil
}
```

I think there should be no difference between the two cases, since the incriminated `return` is in an anonymous function that is identical in both cases.

**Comment by chavacava** [comment]
Hi @olivierperes, thanks for the bug report. 

It is a sort of scoping problem: the rule is interpreting the first `return` as been a `return`of the `f` function and not of the function literal.

I'll try to fix it this week.

**Comment by chavacava** [comment]
@olivierperes you can check the PR #282 that aims to fix this bug

**Comment by olivierperes** [comment]
Thanks!
